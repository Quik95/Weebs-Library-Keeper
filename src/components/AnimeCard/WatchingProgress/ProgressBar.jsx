import React, { memo } from 'react';
import PropTypes from 'prop-types';

// MUI
import Tooltip from '@material-ui/core/Tooltip';

// styles
import clsx from 'clsx';
import useStyles from './WatchingProgress.styles';

const ProgressBar = memo(function ProgressBar({ className, tooltipText }) {
  const classes = useStyles();

  return (
    <Tooltip title={tooltipText}>
      <div className={clsx(classes.progress, classes[className])} />
    </Tooltip>
  );
});

ProgressBar.propTypes = {
  className: PropTypes.string.isRequired,
  tooltipText: PropTypes.string.isRequired,
};

export default ProgressBar;
