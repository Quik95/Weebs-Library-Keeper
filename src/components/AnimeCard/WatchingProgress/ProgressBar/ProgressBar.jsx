import React, { memo } from "react";
import PropTypes from "prop-types";

// MUI
import Tooltip from "@material-ui/core/Tooltip";

// styles
import clsx from "clsx";
import useStyles from "../WatchingProgress.styles";

const ProgressBar = memo(function ProgressBar({
  className,
  tooltipText,
  width = 100
}) {
  const classes = useStyles();

  return (
    <Tooltip title={tooltipText} data-testid="ProgressBar">
      <div
        className={clsx(classes.progress, classes[className])}
        style={{ width: `${width}%` }}
      />
    </Tooltip>
  );
});

ProgressBar.propTypes = {
  className: PropTypes.string.isRequired,
  tooltipText: PropTypes.string.isRequired,
  width: PropTypes.number
};

export default ProgressBar;
