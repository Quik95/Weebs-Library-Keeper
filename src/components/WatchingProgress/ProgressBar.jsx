import React, { memo } from 'react';
import PropTypes from 'prop-types';

// styles
import clsx from 'clsx';
import useStyles from './WatchingProgress.styles';

const ProgressBar = memo(function ProgressBar(props) {
  const classes = useStyles();

  return (
    <div className={classes.progressBarContainer}>
      <div className={clsx(classes['progress--watched'], classes.progress)} />
      <div className={clsx(classes['progress--aired'], classes.progress)} />
      <div className={clsx(classes['progress--total'], classes.progress)} />
    </div>
  );
});

ProgressBar.propTypes = {};

export default ProgressBar;
