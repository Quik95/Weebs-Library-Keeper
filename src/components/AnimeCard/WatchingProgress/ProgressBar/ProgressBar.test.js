import ProgressBar from './ProgressBar';

describe('<ProgressBar/>', () => {
  const wrapper = shallow(
    <ProgressBar
      className="progress--watched"
      tooltipText="Number of episodes you watched"
    />
  );

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Correctly applies classes', () => {
    expect(wrapper.find('div').hasClass(/progress/)).toBe(true);
    expect(wrapper.find('div').hasClass(/progress--watched/)).toBe(true);
  });
});
