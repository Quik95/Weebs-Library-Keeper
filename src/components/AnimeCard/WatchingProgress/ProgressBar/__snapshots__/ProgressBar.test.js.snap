// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar/> Renders correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  data-testid="ProgressBar"
  title="Number of episodes you watched"
>
  <div
    className="makeStyles-progress-2 makeStyles-progress--watched-4"
  />
</WithStyles(ForwardRef(Tooltip))>
`;
